tests/python_package_test/test_arrow.py

```diff
@@ -103,6 +101,29 @@ def test_dataset_construct_fuzzy(
     assert filecmp.cmp(tmp_path / "arrow.txt", tmp_path / "pandas.txt")
 
 
+# -------------------------------------------- FIELDS ------------------------------------------- #
+
+
+@pytest.mark.parametrize("field", ["label", "weight"])
+def test_dataset_construct_fields_fuzzy(field: str):
+    arrow_table = generate_random_arrow_table(3, 1000, 42)
+    arrow_array = generate_random_arrow_array(1000, 42)
+
+    arrow_dataset = lgb.Dataset(arrow_table, **{field: arrow_array})
+    arrow_dataset.construct()
+
+    pandas_dataset = lgb.Dataset(arrow_table.to_pandas(), **{field: arrow_array.to_numpy()})
+    pandas_dataset.construct()
+
+    np_assert_array_equal(arrow_dataset.get_field(field), pandas_dataset.get_field(field))
```

Comment:
Could you please pass through `strict=True` to all these `np_assert_array_equal()` calls? Being able to do that was the main motivation for the added indirection of a helper function in #6108. That leads to `numpy` checking things like dtypes and shape.